File: FE/AL-homepage/src/types/student.ts
export type StudentStatus = 'ACTIVE' | 'ON_LEAVE' | 'WITHDRAWN' | 'DELETED'

export interface Student {
  student_name: string
  branch_id: number
  english_name: string
  student_phone: string
  parent_phone: string
  school: string
  s_year: number
  birthday: string // YYYY-MM-DD
  status: StudentStatus
  student_id: number
  created_at: string // ISO 8601 date string
  updated_at: string | null // ISO 8601 date string or null
  deleted_at: string | null // ISO 8601 date string or null
}

export interface CreateStudentRequest {
  student_name: string
  branch_id: number
  english_name?: string
  student_phone?: string
  parent_phone: string
  school?: string
  s_year: number
  birthday?: string
  status: StudentStatus
}

export interface UpdateStudentRequest {
  english_name?: string
  student_phone?: string
  parent_phone?: string
  school?: string
  s_year?: number
  birthday?: string
  status?: StudentStatus
}

export interface GetAllStudentsParams {
  skip?: number
  limit?: number
  branch_id?: number
  status?: StudentStatus
}

File: BE/avalon/app/models/student.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    DateTime,
    Date,
    Enum,
    ForeignKey
)
from sqlalchemy.sql import func
import enum
from .base import Base

class StudentStatus(str, enum.Enum):
    ACTIVE = "ACTIVE"
    ON_LEAVE = "ON_LEAVE"
    WITHDRAWN = "WITHDRAWN"
    DELETED = "DELETED"

class Student(Base):
    __tablename__ = "students"

    student_id = Column(Integer, primary_key=True, autoincrement=True)
    branch_id = Column(Integer, ForeignKey("branches.branch_id"), nullable=False)
    student_name = Column(String(20), nullable=False)
    english_name = Column(String(100), nullable=True)
    student_phone = Column(String(20), nullable=True)
    parent_phone = Column(String(20), nullable=True)
    school = Column(String(255), nullable=True)
    s_year = Column(Integer, nullable=True)
    birthday = Column(Date, nullable=True)
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=True, onupdate=func.now())
    deleted_at = Column(DateTime, nullable=True)
    status = Column(Enum(StudentStatus), nullable=False, default=StudentStatus.ACTIVE)